Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/measure.h

```diff
@@ -211,6 +211,54 @@ squared_edge_length(typename boost::graph_traits<PolygonMesh>::edge_descriptor e
   return squared_edge_length(halfedge(e, pmesh), pmesh, np);
 }
 
+/**
+  * \ingroup PMP_measure_grp
+  *
+  * Finds the longest edge of a given polygon mesh.
+  *
+  * @tparam PolygonMesh a model of `HalfedgeGraph`
+  * @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+  *
+  * @param pmesh the polygon mesh to which `h` belongs
+  * @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+  *
+  * \cgalNamedParamsBegin
+  *   \cgalParamNBegin{vertex_point_map}
+  *     \cgalParamDescription{a property map associating points to the vertices of `pmesh`}
+  *     \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+  *                    as key type and `%Point_3` as value type}
+  *     \cgalParamDefault{`boost::get(CGAL::vertex_point, pmesh)`}
+  *   \cgalParamNEnd
+  *
+  *   \cgalParamNBegin{geom_traits}
+  *     \cgalParamDescription{an instance of a geometric traits class}
+  *     \cgalParamType{a class model of `Kernel`}
+  *     \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+  *     \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+  *   \cgalParamNEnd
+  * \cgalNamedParamsEnd
+  *
+  * @return the longest edge in `pmesh`. The return type is an `edge_descriptor`.
+  *
+  * @sa `squared_edge_length()`
+  * @sa `longest_border()`
+  */
+
+template<typename PolygonMesh,
+         typename NamedParameters = parameters::Default_named_parameters>
+inline typename boost::graph_traits<PolygonMesh>::edge_descriptor
+longest_edge(const PolygonMesh& pmesh, const NamedParameters& np = parameters::default_values()) {
+
+    auto edge_range = edges(pmesh);
+
+    CGAL_assertion(edge_range.size() > 0);
```

Comment:
`boost::graph_traits<PolygonMesh>::null_edge()` could be returned if the mesh is empty (testing `edge_range.begin()==edge_range.end()`)

Reply:
@sloriot  Thank you for the feedback. I am a bit confused. I assume you mean that I should replace the assertion with a condition `(edge_range.begin()==edge_range.end())` and return `boost::graph_traits<PolygonMesh>::null_edge()`. Is that right?
I also added a check on the returned variable from the `max_element()` before dereferencing it for safety.  I added these updates and your suggestions to new commit.